### ISSUE TITLE
Kappa for SG 

### ISSUE THREAD

**Issue by tiramisuq** [Intel-bigdata/imllib-spark#23]
Hi there,
Here is a confussion of kappa for SG: as per the paper, when gradient is calculated with -label / (1 + expnyt) where expnyt = math.exp(label * t), however in code FieldAwaredFactorizationMachine.scala: line 312, expnyt = math.exp(-label * t). I think it is mistaken as log loss.
The optimization will not converge....
Regards

**Comment by kapilkd13** [comment]
I don't think this is an issue here, as kappa is defined as,` val kappa = -label * expnyt / (1 + expnyt)` where expnyt is `val expnyt = math.exp(-label * t)`. Taking exponent common from numerator and denominator we get the kappa value as mentioned in the paper. I think this is done to avoid calculating inverse exponynt as req by kappa expression.
